codec/websocket/definitions.go

```diff
@@ -357,6 +360,24 @@ type Stream interface {
 	// ControlCallback returns the control callback set with SetControlCallback.
 	ControlCallback() ControlCallback
 
+	// SetUpgradeRequestCallback sets a function that will be invoked during the handshake
+	// just before the upgrade request is sent.
+	//
+	// The caller must not perform any operations on the stream in the provided callback.
+	SetUpgradeRequestCallback(callback UpgradeRequestCallback)
```

Comment:
I'm guessing this is here in order to provide the `http.Request` to the caller for potential modification... which for the handshake means adding more headers. If so, we already have this mechanism of adding extra headers to the handshake, see [here](https://github.com/talostrading/sonic/blob/master/codec/websocket/stream.go#L635).

Reply:
It's for logging the request (at least for my use case).